docs/devel/api_changes.md

```diff
@@ -0,0 +1,237 @@
+# So you want to change the API?
+
+The Kubernetes API has two major components - the internal API structures and
+the versioned API structures.  The versioned APIs are intended to be stable,
+while the internal API is implemented to best reflect the needs of the
+Kubernetes code itself.  Clients should consume the versioned APIs exclusively.
+
+What this means for API changes is that you have to be somewhat thoughtful in
+how you approach changes, and that you have to touch a number of pieces to make
+a complete change.  This document aims to guide you through the process - not
+all API changes will need all of these steps.
+
+## On compatibility
+
+Before talking about API changes, it is worthwhile to clarify what we mean by
+API compatibility.  An API change is considered compatible if it:
+   * adds new functionality that is not required for correct behavior
+   * does not change existing semantics
+   * does not change existing defaults
+
+Put another way:
+
+1. An API call (e.g. a structure POSTed to a REST endpoint) that worked before
+   your change must work the same after your change.
+2. An API call that uses your change must work correctly against servers that
```

Comment:
Good.  Point.  I am still wrestling with how to express this succinctly.

2. Any API call that does not use your change must work the same after your change.
3. Any API call that uses your change must not crash or otherwise degrade servers that do not include your change